addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -111,32 +121,24 @@ def action_cancel(self):
             requisition.purchase_ids.button_cancel()
             for po in requisition.purchase_ids:
                 po.message_post(body=_('Cancelled by the agreement associated to this quotation.'))
-        self.write({'state': 'cancel'})
+        self.state = 'cancel'
 
     def action_in_progress(self):
         self.ensure_one()
         if not self.line_ids:
-            raise UserError(_("You cannot confirm agreement '%s' because there is no product line.", self.name))
-        if self.type_id.quantity_copy == 'none' and self.vendor_id:
+            raise UserError(_("You cannot confirm agreement '%(agreement)s' because it does not contain any product lines.", agreement=self.name))
+        if self.requisition_type == 'blanket_order':
             for requisition_line in self.line_ids:
                 if requisition_line.price_unit <= 0.0:
-                    raise UserError(_('You cannot confirm the blanket order without price.'))
+                    raise UserError(_('You cannot confirm a blanket order without any prices.'))
```

Comment:
Mm, sorry now that I look at this again, I didn't tell you the correct English for this.

```suggestion
                    raise UserError(_('You cannot confirm a blanket order with lines missing a price.'))
```

Same thing for the usererror below it

`raise UserError(_('You cannot confirm a blanket order with lines missing a quantity.'))`